src/Plutip/Types.purs

```diff
@@ -197,27 +231,116 @@ instance DecodeAeson StopClusterResponse where
 -- | distribution for wallets.
 -- | Number of wallets in distribution specification matches the number of
 -- | wallets provided to the user.
-class UtxoDistribution distr wallets | distr -> wallets, wallets -> distr where
+class UtxoDistribution distr wallets | distr -> wallets where
   encodeDistribution :: distr -> Array (Array UtxoAmount)
-  decodeWallets :: Array PrivateKeyResponse -> Maybe wallets
+  decodeWallets :: distr -> Array PrivateKeyResponse -> Maybe wallets
+  keyWallets :: Proxy distr -> wallets -> Array KeyWallet
```

Comment:
Couldn't find a way to get this to type check without the proxy. Giving type annotations where the method is used like `keyWallets :: blah` didn't work

Reply:
a Proxy is OK